`no-unexpected-multiline` false positive (eslint 2.0.0)

Using the latest eslint (2.0.0-rc.0).

Rules:

```json
{
  "rules": {
    "no-unexpected-multiline": 2
  }
}
```

Code:

```js
(
  function () {}
)[1]
```

Results in the following false positive:

```
  3:2  error  Unexpected newline between object and [ of property access  no-unexpected-multiline
```

This does not trigger an error in eslint v1.10.3.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30520351-no-unexpected-multiline-false-positive-eslint-2-0-0?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/lib/rules/no-unexpected-multiline.js
+++ b/lib/rules/no-unexpected-multiline.js
@@ -23,13 +23,17 @@ module.exports = function(context) {
      * @private
      */
     function checkForBreakAfter(node, msg) {
-        var paren = context.getTokenAfter(node);
-        while (paren.value === ")") {
-            paren = context.getTokenAfter(paren);
+        var nodeExpressionEnd = node;
+        var openParen = context.getTokenAfter(node);
+
+        // Move along until the end of the wrapped expression
+        while (openParen.value === ")") {
+            nodeExpressionEnd = openParen;
+            openParen = context.getTokenAfter(nodeExpressionEnd);
         }
 
-        if (paren.loc.start.line !== node.loc.end.line) {
-            context.report(node, paren.loc.start, msg, { char: paren.value });
+        if (openParen.loc.start.line !== nodeExpressionEnd.loc.end.line) {
+            context.report(node, openParen.loc.start, msg, { char: openParen.value });
         }
     }
 

--- a/tests/lib/rules/no-unexpected-multiline.js
+++ b/tests/lib/rules/no-unexpected-multiline.js
@@ -25,6 +25,7 @@ ruleTester.run("no-unexpected-multiline", rule, {
         "\"abc\\\n(123)\"",
         "var a = (\n(123)\n)",
         "f(\n(x)\n)",
+        "(\nfunction () {}\n)[1]",
         {
             code: "let x = function() {};\n   `hello`",
             parserOptions: { ecmaVersion: 6 }

